lib/modules/datasource/kubernetes-api/mapping.ts

```diff
@@ -0,0 +1,118 @@
+export const KubernetesApiVersions: Record<string, string[]> = {
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-16
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#networkpolicy-v116
+  NetworkPolicy: ['extensions/v1beta1', 'networking.k8s.io/v1'],
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#daemonset-v116
+  DaemonSet: ['extensions/v1beta1', 'apps/v1beta2', 'apps/v1'],
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#deployment-v116
+  Deployment: ['extensions/v1beta1', 'apps/v1beta1', 'apps/v1beta2', 'apps/v1'],
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#statefulset-v116
+  StatefulSet: ['apps/v1beta1', 'apps/v1beta2', 'apps/v1'],
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#replicaset-v116
+  ReplicaSet: ['extensions/v1beta1', 'apps/v1beta1', 'apps/v1beta2', 'apps/v1'],
+  // https://kubernetes.io/docs/reference/using-api/deprecation-guide/#psp-v116
+  PodSecurityPolicy: ['extensions/v1beta1', 'policy/v1beta1'],
```

Comment:
those will be removed in V1.25

Reply:
True, but upgrading to `policy/v1beta1` is as far as Renovate can go. The migration guide[^1] requires manual intervention.

[^1]: https://kubernetes.io/docs/tasks/configure-pod-container/migrate-from-psp/